In [2]:
from sklearn.linear_model import LinearRegression 
import numpy as np
import json

regression_model = LinearRegression()



Features are:
* Is doctor reply: `0` or `1`
* Number of votes: `int > 0`
* Number of informative entities `int > 0 `
* Number of informative entities shared with query `int > 0 `
* Length more than 150 `0` or `1`
* Is same category `0` or `1`
* Is same thread `0` or `1`


In [3]:
with open("d:/downloads/json/healthboards/trac/training_data.json", "r", encoding="utf8") as file:
    training_data = json.load(file)

print(len(training_data))

with open("d:/downloads/json/healthboards/trac/targets.json", "r", encoding="utf8") as file:
    targets = json.load(file)

print(len(targets))

x = np.array(training_data)
y = np.array(targets)

1056352
1056352


In [4]:
regression_model.fit(x, y)

LinearRegression(copy_X=True, fit_intercept=True, n_jobs=None, normalize=False)

In [5]:
weights = regression_model.coef_.tolist()[0]

for i in weights:
    print("%5.3f" % i)
# print(regression_model.get_params())

# test_sample = np.array([[0, 0, 2, 2, 1, 1, 0]])
# regression_model.predict(test_sample)

0.000
0.000
2.112
-1.909
-1.127
15.491
0.772
5.999
52.523





| -| Doctor reply | votes (helpful) | votes (support) | votes (thank you) | Medical entities | Same entities | Length factor | Same category | Same thread |
|-|--------------|-----------------|-----------------|-------------------|------------------|---------------|---------------|---------------|-------------|
eHealthForum| -13.9  | 0.819 | 0| 0  | 0.187 | 2.77 | 0.620  | 4.64  | 10.7  |
HealthBoards | 0.0 | 0.0 | 2.112 | -1.9 | -1.12 | 15.5 | 0.77 | 5.99 | 52.52 |

Raw (ehealthforum):
-13.965
0.819
0.000
0.000
0.187
2.771
0.621
4.644
10.759

Raw (healthboards): 
0.000
0.000
2.112
-1.909
-1.127
15.491
0.772
5.999
52.523